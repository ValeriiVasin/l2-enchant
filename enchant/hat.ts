import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const hatEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Destroy,
  enchantItem: Item.EvaScrollEnchantHairAccessory,
  chances: [100, 100, 100, 50, 40, 40, 8, 8, 5, 2],
  items: itemsRange(Item.Hat, Item.Hat_10),
});
