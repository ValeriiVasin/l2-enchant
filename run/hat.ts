import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.Hat, [Item.Hat_4, Item.Hat_5, Item.Hat_6, Item.Hat_7], 100_000);
