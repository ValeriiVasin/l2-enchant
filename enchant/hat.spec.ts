import { hatEnchantMap } from './hat';

it('enchant map: hat', () => {
  expect(hatEnchantMap).toMatchSnapshot();
});
