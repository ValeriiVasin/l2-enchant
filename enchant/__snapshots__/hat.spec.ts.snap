// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: hat 1`] = `
Map {
  "Головной Убор" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Головной Убор +1",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +1" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +1",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Головной Убор +2",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +2" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +2",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Головной Убор +3",
      },
      Object {
        "chance": 0,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +3" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +3",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Головной Убор +4",
      },
      Object {
        "chance": 50,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +4" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +4",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Головной Убор +5",
      },
      Object {
        "chance": 60,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +5" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +5",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Головной Убор +6",
      },
      Object {
        "chance": 60,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +6" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +6",
    "results": Array [
      Object {
        "chance": 8,
        "item": "Головной Убор +7",
      },
      Object {
        "chance": 92,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +7" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +7",
    "results": Array [
      Object {
        "chance": 8,
        "item": "Головной Убор +8",
      },
      Object {
        "chance": 92,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +8" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +8",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Головной Убор +9",
      },
      Object {
        "chance": 95,
        "item": "Nothing",
      },
    ],
  },
  "Головной Убор +9" => Object {
    "cost": "Свиток Евы: Модифицировать Головной Убор",
    "item": "Головной Убор +9",
    "results": Array [
      Object {
        "chance": 2,
        "item": "Головной Убор +10",
      },
      Object {
        "chance": 98,
        "item": "Nothing",
      },
    ],
  },
}
`;
